activejob/lib/active_job/enqueuing.rb

```diff
@@ -59,10 +71,12 @@ def enqueue(options = {})
           queue_adapter.enqueue self
         end
 
-        successfully_enqueued = true
+        self.successfully_enqueued = true
+      rescue EnqueueError => e
+        self.enqueue_error = e
       end
 
-      if successfully_enqueued
+      if self.successfully_enqueued?
```

Comment:
```suggestion
      if successfully_enqueued?
```

Reply:
Updated as requested. Thanks for looking :)